PROD-2223 Various BE cleanup for property specific messaging

eastandwestwind opened the pull request:
Closes https://ethyca.atlassian.net/browse/PROD-2223

### Description Of Changes

1. Fix the issue where PATCHing a template will remove properties.
2. Adds validation to prevent enabling a template with no properties
3. Exposes minimal messaging template in public property response.


### Code Changes

* [ ] Adjust CRUD logic to support desired functionality described above
* [ ] Update tests

### Steps to Confirm

* [ ] Full test coverage, if you wish to test manually, run plus pointing to the latest commit of this PR
* [ ] Run https://github.com/ethyca/fides/pull/5005 locally, confirm you're able to enable/disable templates on the table and any properties previously linked are still preserved. 
* [ ] Confirm that you cannot enable a template with no properties
* [ ] Confirm that GET [/api/v1/plus/properties](http://localhost:8080/docs#/Plus/get_properties_api_v1_plus_properties_get) returns list of messaging templates

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] if there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* If there are any database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!

galvana reviewed (commented):
Nice test coverage, works as expected! I just left a few comments.

There's one bug I found that might be better logged somewhere else but it came up during testing. Now that we added `messaging_templates` to the `Property` schema, I am not able to create a property from the UI because the old forms aren't passing back the `messaging_templates` value. I think this is a simple fix, @lucanovera had to do this when I added `paths`.

Second thing, that is also front-end related, can we make it clearer to the user that the reason the messaging template couldn't be enabled is because it didn't have a property? Right now I get the more generic "This message cannot be enabled because another message already exists with the same configuration. Change the property to enable this message."

These two aren't things to fix in this ticket but I just wanted to at least call them out!

eastandwestwind commented on tests/ops/schemas/test_property_schema.py:
```diff
@@ -17,6 +17,7 @@ def test_valid_paths(self, paths):
             "name": "Test Property",
             "type": PropertyType.website,
             "experiences": [],
+            "messaging_templates": [],
```
good point, I'll check this!

eastandwestwind commented:
@galvana re: 

> can we make it clearer to the user that the reason the messaging template couldn't be enabled is because it didn't have a property? Right now I get the more generic "This message cannot be enabled because another message already exists with the same configuration. Change the property to enable this message."

We can def separate out 2 different messages for the following error cases when trying to enable a property:
1. Another template has the same property and template type
2. There are no properties added to the template at all

We'll just need to confirm with @jack-gale-ethyca what the copy should read for each of these cases. 

It'll also take a little FE parsing of the BE err codes to ensure we're choosing the correct message to display for each case, which we can tackle in a separate ticket.

eastandwestwind commented:
@galvana I've fixed both of your call-outs in this PR:
1. Fixed the Property form such that we can add / edit / delete properties with no issues
2. Updated the err messages from the BE such that we can use them directly on the FE, and this way we can be more specific with the messages 🎉

galvana reviewed (approved):
Thanks for taking the extra time to improve the messaging, approved!
